### PR TITLE
Add support for creating PR's in ghes and behind proxy

### PR DESCRIPTION
**Pull request by sethumadhav07**
This PR addresses the issue #39

### CODE REVIEW

File: src/index.ts
Comment (thomaseizinger) [inline src/index.ts:16]: Is this some kind of standardized env variable? I think I'd prefer this to be a input because it is easier to test.

File: src/getInputs.ts
Comment (thomaseizinger) [inline src/getInputs.ts:18]: Instead of reading this as an input, I think we can pull the correct value out of the environment: https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Left some comments inline.

File: src/getInputs.ts
Comment (sethumadhav07) [reply src/getInputs.ts]: addressed this.

File: src/index.ts
Comment (sethumadhav07) [reply src/index.ts:14]: yes it is standardized env variable. i have seen various other github actions doing very similar.

Comment (thomaseizinger) [reply src/index.ts:14]: Okay, can we document this in the README then please?

Comment (sethumadhav07) [reply src/index.ts:14]: sure will do.

**Comment by thomaseizinger** [comment]
Hmm, I never tried running these CI scripts from a fork, only from my own branches.

What is your appetite for trying fix this? I think it should be doable by adding the fork that exists under your username and explicitly pushing to that so we can open a pull-request.

It is fine if you don't want to but I also can't promise that I'll find time very soon to try and play around with it.

**Comment by sethumadhav07** [comment]
can you approve the above PR and see if that fixes it?

Not sure what you mean by - "I think it should be doable by adding the fork that exists under your username and explicitly pushing to that so we can open a pull-request."?

i am actually pushing to my fork only.

**Comment by thomaseizinger** [comment]
> can you approve the above PR and see if that fixes it?
> 
> Not sure what you mean by - "I think it should be doable by adding the fork that exists under your username and explicitly pushing to that so we can open a pull-request."?
> 
> i am actually pushing to my fork only.

Sorry, I should have been more clear.
The CI scripts actually use the action to open pull requests: https://github.com/thomaseizinger/create-pull-request/pulls?q=is%3Apr+sort%3Aupdated-desc+is%3Aclosed+author%3Aapp%2Fgithub-actions, effectively integration testing the code before it is being merged. That only worked though because I was developing in branches (and not forks) and hence the token provided in `GITHUB_TOKEN` had enough permissions to push to a branch.

You are the first one to contribute which shows that my CI scripts don't work with forks :sweat_smile: 

What I meant with "I think it should be doable by adding the fork that exists under your username and explicitly pushing to that so we can open a pull-request" is that the [CI scripts](https://github.com/thomaseizinger/create-pull-request/blob/13c98abb44c384d2978a9cd564b193221f116b83/.github/workflows/CI.yml#L25) could be modified to try and push to a remote that is under the username of whoever triggered the workflow. There is a `git_url` field inside the `pull_request` event payload https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request that we would need to use.
Once a remote branch is pushed, the create-pull-request action can be run again.

However, maybe this kind of integration testing is also overkill though and should just be removed :)

**Comment by sethumadhav07** [comment]
> > can you approve the above PR and see if that fixes it?
> > Not sure what you mean by - "I think it should be doable by adding the fork that exists under your username and explicitly pushing to that so we can open a pull-request."?
> > i am actually pushing to my fork only.
> 
> Sorry, I should have been more clear.
> The CI scripts actually use the action to open pull requests: https://github.com/thomaseizinger/create-pull-request/pulls?q=is%3Apr+sort%3Aupdated-desc+is%3Aclosed+author%3Aapp%2Fgithub-actions, effectively integration testing the code before it is being merged. That only worked though because I was developing in branches (and not forks) and hence the token provided in `GITHUB_TOKEN` had enough permissions to push to a branch.
> 
> You are the first one to contribute which shows that my CI scripts don't work with forks 😅
> 
> What I meant with "I think it should be doable by adding the fork that exists under your username and explicitly pushing to that so we can open a pull-request" is that the [CI scripts](https://github.com/thomaseizinger/create-pull-request/blob/13c98abb44c384d2978a9cd564b193221f116b83/.github/workflows/CI.yml#L25) could be modified to try and push to a remote that is under the username of whoever triggered the workflow. There is a `git_url` field inside the `pull_request` event payload https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request that we would need to use.
> Once a remote branch is pushed, the create-pull-request action can be run again.
> 
> However, maybe this kind of integration testing is also overkill though and should just be removed :)

Unfortunately, I don't currently have time for this. i just want this PR to be merged so that we can use this github action.
If you want me to remove the three tests that are failing, i can quickly remove it. Otherwise we can run tests only when the PRs are not from fork until someone gets time to address it?

**Comment by thomaseizinger** [comment]
> > > can you approve the above PR and see if that fixes it?
> > > Not sure what you mean by - "I think it should be doable by adding the fork that exists under your username and explicitly pushing to that so we can open a pull-request."?
> > > i am actually pushing to my fork only.
> > 
> > 
> > Sorry, I should have been more clear.
> > The CI scripts actually use the action to open pull requests: [thomaseizinger/create-pull-request/pulls (sort:updated-desc is:closed author:app/github-actions)](https://github.com/thomaseizinger/create-pull-request/pulls?q=is%3Apr+sort%3Aupdated-desc+is%3Aclosed+author%3Aapp%2Fgithub-actions), effectively integration testing the code before it is being merged. That only worked though because I was developing in branches (and not forks) and hence the token provided in `GITHUB_TOKEN` had enough permissions to push to a branch.
> > You are the first one to contribute which shows that my CI scripts don't work with forks sweat_smile
> > What I meant with "I think it should be doable by adding the fork that exists under your username and explicitly pushing to that so we can open a pull-request" is that the [CI scripts](https://github.com/thomaseizinger/create-pull-request/blob/13c98abb44c384d2978a9cd564b193221f116b83/.github/workflows/CI.yml#L25) could be modified to try and push to a remote that is under the username of whoever triggered the workflow. There is a `git_url` field inside the `pull_request` event payload [docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request) that we would need to use.
> > Once a remote branch is pushed, the create-pull-request action can be run again.
> > However, maybe this kind of integration testing is also overkill though and should just be removed :)
> 
> Unfortunately, I don't currently have time for this. i just want this PR to be merged so that we can use this github action.
> If you want me to remove the three tests that are failing, i can quickly remove it. Otherwise we can run tests only when the PRs are not from fork until someone gets time to address it?

I will take your commits and push them to a branch on the repo. That should be good enough for now. Thanks for the contribution!

**Comment by thomaseizinger** [comment]
Closed in favor of #41.

**Pull request closed without merging**
